lib/collection/src/shards/transfer/mod.rs

```diff
@@ -79,6 +80,9 @@ pub enum ShardTransferMethod {
 /// Interface to consensus for shard transfer operations.
 #[async_trait]
 pub trait ShardTransferConsensus: Send + Sync {
+    /// Get current CPU budget.
+    fn cpu_budget(&self) -> CollectionResult<CpuBudget>;
```

Comment:
how is `cpu_budget` a part of `ShardTransferConsensus`?

Reply:
Where we access it [here](https://github.com/qdrant/qdrant/blob/4122b7e49cae7916db27f47b47faa9616c4ff213/lib/collection/src/collection/shard_transfer.rs#L95), we can simply do `self.optimizer_cpu_budget`. I've updated it and removed CPU budget from the trait. Thanks for this insight @generall!